framework/src/main/java/org/tron/core/net/service/adv/AdvService.java

```diff
@@ -368,7 +374,16 @@ public void sendInv() {
         }
         if (key.equals(InventoryType.BLOCK)) {
           value.sort(Comparator.comparingLong(value1 -> new BlockId(value1).getNum()));
-          peer.sendMessage(new InventoryMessage(value, key));
+          if (testStopInv && peer.isNotActiveTooLong()
+              && peer.getMaliciousFeature().getStopBlockInvTime() == -1) {
+            //if peer is not active for too long, test if peer will broadcast block inventory to me
+            //after I stop broadcasting block inventory to it
+            peer.getMaliciousFeature().setStopBlockInvTime(System.currentTimeMillis());
+            invCheckExecutor.schedule(() -> peer.getMaliciousFeature().updateBadFeature4(),
```

Comment:
Why use a scheduler to do this?

Reply:
We check that if we receive some inventory from peer in 10 seconds, it is a timer. It is scheduled only once.